### DISCUSSION TITLE
Use different shell for nx:run-commands executor?

### DISCUSSION THREAD

**Discussion by ls-bit** [nrwl/nx#17961] [General]
We are using pnpm and have [script-shell](https://pnpm.io/cli/run#script-shell) configured to use git bash on Windows. This allows us to use common bash syntax in package.json scripts even on Windows.

[nx:run-commands](https://nx.dev/packages/nx/executors/run-commands) executor seems to always use cmd.exe to run dependent tasks defined by [task pipeline](https://nx.dev/concepts/task-pipeline-configuration), which does not work with our scripts. So we would like to configure shell to use git bash.

Is there any way to configure the shell used by nx:run-commands executor?

**Comment by phaze-phusion** [comment]
Our team also use pnpm and use bash scripts everywhere. For us Nx defaults to using `/bin/sh`, with nowhere to configure it. We have developers on both Ubuntu and Mac. Everyone has bash installed even if they use a different shell (bash, zsh or fish). Even our CI pipelines use bash.

I've found that using `/bin/bash -c "..."` in the command property of nx:run-commands allows us to use bash syntax. But that requires careful escaping of quotation marks and even then it can get confusing because of JSON's own double quotation marks.

For anyone wanting to go this route here is a simple example:
```
"arbitrary": {
  "executor": "nx:run-commands",
  "options": {
    "commands": [
      {
        "command": "/bin/bash -c 'opt=\"some text\"; [[ $opt == \"some text\" ]] && echo true || echo false'"
      }
    ]
  }
}
```

**Reply by phaze-phusion** [reply]
Update:
Looking at the [run-commands source](https://github.com/nrwl/nx/blob/master/packages/nx/src/executors/run-commands/run-commands.impl.ts), nx:run-command uses Node's `child_process.exec()` method which has an option to set the shell. That option just needs to be made available in the schema and passed on to the exec method.

According to [NodeJS docs](https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback) the shell defaults to `/bin/sh` on Unix, and `process.env.ComSpec` on Windows.

**Reply by landon-buttars-wgu** [reply]
This issue is particularly problematic for lean environments that may not have sh installed. Has there been an update since?

**Comment by seth-peters-apiture** [comment]
We would also like to be able to customize the shell used by nx:run-commands without having to use `/bin/bash -c "..."`

**Comment by DavidVollmers** [comment]
Please implement this option...
